app/components/FolderContentScreen.tsx

```diff
@@ -77,7 +82,32 @@ const FolderContentScreen = (props: NativeStackScreenProps<RootStackParamList, "
             });   
         }
         catch {
-            Alert.alert("File upload failed!");
+            Alert.alert("Failed to upload file!");
+        }
+    }
+
+    const addFolderHandler = async () => {
+        try {
+            await createFolder(`${props.route.params?.folderURL}/${inputText}`).then((status) => {
```

Comment:
We should escape special characters. Didn't test everything but for example if you put in a [ the server doesn't create a folder and the user also doesn't get feedback that no folder was created

Reply:
Actually for me nextcloud handles folders named `[` well. But yes we don't want such filenames allowed therefore users can now only create folder names with lating letters, numbers, "-", "_" and spaces. When they enter something else they will see a error message explaining whats allowed.